### ISSUE TITLE
Build with static configuration not working

### ISSUE THREAD

**Issue by ksRaffaelSenn** [fluent/fluent-bit#1242]
## Bug Report

**Describe the bug**

CMake building with static configuration like it is explained in https://docs.fluentbit.io/manual/installation/build_static_configuration is not working. 

**To Reproduce**
- Error message:
```
CMake Error at CMakeLists.txt:387 (add_subdirectory):
  add_subdirectory given source "gen_static_conf" which is not an existing
  directory
```
- Log-Files: [Logs.zip](https://github.com/fluent/fluent-bit/files/3034037/Logs.zip)
- Steps to reproduce the problem:
  - Get the 1.0.6 release of fluent-bit
  - Create a directory configs
  - Create a config within the directory
  - change directory to build
  - run:
  ```
  cmake -DFLB_STATIC_CONF=configs ..
  ```

**Expected behavior**

It should build fluent-bit with the plugins and properties given in the config file

**Your Environment**

* Version used: 1.0.6
* Configuration: [configs.zip](https://github.com/fluent/fluent-bit/files/3033972/configs.zip)
* Environment name and version: I'm building this in a docker container. My Dockerfile is 
[Dockerfile.zip](https://github.com/fluent/fluent-bit/files/3033994/Dockerfile.zip)
* Server type and version: I'm using docker 18.09.4
* Operating System and version: Ubuntu 18.04.2. In the Container, I'm using debian:stretch as base image.

**Additional context**

We want to use fluent-bit in an embedded device. We build the os and all apps for the device from source. It would be nice if we could configure fluent-build while building with a static configuration, so we can minimize memory footprint and dependencies.

**Comment by edsiper** [comment]
did you created the __configs/__ directory in the root path or inside build ?, looks like you are positioned in build/ but your __configs/__ one directory level up.

**Comment by ksRaffaelSenn** [comment]
I tried putting the configs driectory in the build driectory. It returned the same error. I also tried relative and absolute paths. It always showed the same error

**Comment by edsiper** [comment]
thanks for reporting the problem, a missing CMake directory was not included in the build system.

Fixed by 8dd6b34e.

note: you can try building using GIT master

**Comment by ksRaffaelSenn** [comment]
Thank you for your fix. Fluent-bit is building now. A note for those who also have trouble building with a static configuration: You also need flex (https://packages.debian.org/de/stretch/flex) and bison (https://packages.debian.org/de/stretch/bison) for the building process.

For our case it would be nice if fluent-bit would only build the dependencies and plugins it needs for the static configuration. It seems like it is building the default build configuration and generating a binary that is bigger than 5MB. 

The binary is also not running. I'm getting the following output:
```
Fluent Bit v1.1.0
Copyright (C) Treasure Data

[engine] caught signal (SIGSEGV)
#0  0x7fb9c6ae7a03      in  ???() at ???:0
#1  0x55bbd1c1c253      in  ???() at ???:0
#2  0x55bbd1c1c816      in  ???() at ???:0
#3  0x55bbd1c1c9f2      in  ???() at ???:0
#4  0x55bbd1bf8dd9      in  ???() at ???:0
#5  0x55bbd1bf962e      in  ???() at ???:0
#6  0x7fb9c6a892e0      in  ???() at ???:0
#7  0x55bbd1bf80e9      in  ???() at ???:0
#8  0xffffffffffffffff  in  ???() at ???:0
Aborted (core dumped)
```

Unfortunately I can't provide the dumped core file, because this seems to be a little tricky in a docker container.

Should I create a new Issue for this?

**Comment by edsiper** [comment]
@ksRaffaelSenn 

I will fix the dep of Flex/Bison shortly. 

Please open a new issue to track the problem you are facing now.

thanks again.

**Comment by vixns** [comment]
Please reopen this issue, the build is still broken, the .h files are not generated by cmake.
I was able to build a working binary by manually running `xxd-c -i fluent-bit.conf -o ${fluentbit_src_path}/include/fluent-bit/conf/fluent-bit.conf.h -s fluent-bit.conf` before running `make`
We may have a chicken/egg problem here, as the xxd-c does not exists yet when running cmake. We may have to delay the headers files generation after creating the `build/bin/xxd-c` binary.

**Comment by edsiper** [comment]
@vixns 

thanks for pointing out the issue. Indeed, the problem is only caught upon a fresh build. I've pushed a fix for the problem: 1a6370969e1590d26847e95e1bf42ee6e2c80e13. 

I was not able to break it again, in case you find any problem please open a new ticket and reference this one.

thanks again.
